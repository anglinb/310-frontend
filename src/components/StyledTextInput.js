import React from 'react'
import { 
  StyleSheet,
  TextInput,
  Text,
  View,
} from 'react-native'
import config from '../config'

export default class StyledTextInput extends React.Component {

  constructor(props) {
    super(props)

    this.renderLabel = this.renderLabel.bind(this)
  }

  renderLabel() {
    if (this.props.labelText) {
      return (
        <Text style={textInputStyles.label}>{this.props.labelText}</Text>
      )
    }
    return null
  }

  render() {
    return (
      <View style={textInputStyles.wrapper}>
        { this.renderLabel() } 
        <TextInput style={StyleSheet.flatten([textInputStyles.textInput, this.props.style])} {...this.props} ></TextInput>
      </View>
    )
  }
}

const textInputStyles = StyleSheet.create({
  wrapper: {
    display: 'flex',
    alignItems: 'stretch',
    borderColor: 'blue',
    borderWidth: 2,
    paddingTop: 10,
  },
  label: {
    color: config.darkText,
    fontFamily: config.font,
    fontWeight: 'bold',
    fontSize: 18,
  },
  textInput: {
    borderRadius: 8,
    height: 36,

    color: config.darkText,
    fontFamily: config.font,
    fontWeight: 'bold',
    fontSize: 18,
    padding: 8, 
    backgroundColor: config.darkBackground,
    borderColor: 'red',
    borderWidth: 2,
  } 
})