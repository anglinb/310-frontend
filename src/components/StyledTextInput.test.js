import React from 'react';
import StyledTextInput from './StyledTextInput';

import { shallow } from 'enzyme';
import sinon from 'sinon';

import renderer from 'react-test-renderer';

it('renders without crashing w/o label', () => {
  const rendered = renderer.create(
    <StyledTextInput
     />
  ).toJSON()
  expect(rendered).toBeTruthy()
  expect(rendered).toMatchSnapshot()
})

it('renders without crashing w/ label', () => {
  const rendered = renderer.create(
    <StyledTextInput
      labelText={`Label`}
     />
  ).toJSON()
  expect(rendered).toBeTruthy()
  expect(rendered).toMatchSnapshot()
})