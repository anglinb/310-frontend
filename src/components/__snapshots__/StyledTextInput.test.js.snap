// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing w/ label 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "display": "flex",
      "paddingTop": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#676E6E",
        "fontFamily": "Avenir Next",
        "fontSize": 18,
        "fontWeight": "bold",
      }
    }
  >
    Label
  </Text>
  <TextInput
    labelText="Label"
    style={
      Object {
        "backgroundColor": "#D8D8D8",
        "borderRadius": 8,
        "color": "#676E6E",
        "fontFamily": "Avenir Next",
        "fontSize": 18,
        "fontWeight": "bold",
        "height": 36,
        "padding": 8,
      }
    }
  />
</View>
`;

exports[`renders without crashing w/o label 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "display": "flex",
      "paddingTop": 10,
    }
  }
>
  <TextInput
    style={
      Object {
        "backgroundColor": "#D8D8D8",
        "borderRadius": 8,
        "color": "#676E6E",
        "fontFamily": "Avenir Next",
        "fontSize": 18,
        "fontWeight": "bold",
        "height": 36,
        "padding": 8,
      }
    }
  />
</View>
`;
